riscv/insns/cbo_clean.h

```diff
@@ -0,0 +1,11 @@
+require_extension(EXT_ZICBOM);
+if (((STATE.prv != PRV_M) && (get_field(STATE.menvcfg->read(), MENVCFG_CBCFE) == 0)) ||
+    ((STATE.prv == PRV_U && !STATE.v) && (get_field(STATE.senvcfg->read(), SENVCFG_CBCFE) == 0)))
+  throw trap_illegal_instruction(insn.bits());
+else if (((STATE.prv == PRV_S && STATE.v) && (get_field(STATE.henvcfg->read(), HENVCFG_CBCFE) == 0)) ||
+    ((STATE.prv == PRV_U && STATE.v) && ((get_field(STATE.henvcfg->read(), HENVCFG_CBCFE) == 0) ||
+    (get_field(STATE.senvcfg->read(), SENVCFG_CBCFE) == 0))))
+  throw trap_virtual_instruction(insn.bits());
```

Comment:
This same check appears in all 4 instructions. Can this be replaced with a macro or function call `require_zicbom(MENVCFG_CBCFE)`? (This assumes that `MENVCFG_CBCFE == SENVCFG_CBCFE == HENVCFG_CBCFE` which should probably be asserted.)

Reply:
maybe we can use require_zicbom(CBCFE) without assert.

Reply:
I have add ZICBOM_OP to support this:
```
#define ZICBOM_OP(field, BODY) \
  require_extension(EXT_ZICBOM); \
  reg_t m##field = get_field(STATE.menvcfg->read(), MENVCFG_##field); \
  reg_t s##field = get_field(STATE.senvcfg->read(), SENVCFG_##field); \
  reg_t h##field = get_field(STATE.henvcfg->read(), HENVCFG_##field); \
  if (((STATE.prv != PRV_M) && (m##field == 0)) || \
      ((STATE.prv == PRV_U && !STATE.v) && (s##field == 0))) \
    throw trap_illegal_instruction(insn.bits()); \
  else if (STATE.v && ((h##field == 0) || \
                       ((STATE.prv == PRV_U) && (s##field == 0)))) \
    throw trap_virtual_instruction(insn.bits()); \
  else { \
    BODY \
  }
```